controllers/clusters/opensearch_controller.go

```diff
@@ -790,6 +806,57 @@ func (r *OpenSearchReconciler) newWatchBackupsJob(o *v1beta1.OpenSearch) schedul
 	}
 }
 
+func (r *OpenSearchReconciler) newUsersCreationJob(o *v1beta1.OpenSearch) scheduler.Job {
+	logger := log.Log.WithValues("component", "openSearchUsersCreationJob")
+
+	return func() error {
+		ctx := context.Background()
+
+		err := r.Get(ctx, types.NamespacedName{
+			Namespace: o.Namespace,
+			Name:      o.Name,
+		}, o)
+		if err != nil {
+			if k8serrors.IsNotFound(err) {
+				logger.Info("OpenSearch cluster resource is not found. Closing user creation job...")
+				r.Scheduler.RemoveJob(o.GetJobID(scheduler.UserCreator))
+				return nil
+			}
+			return err
+		}
+
+		if o.Status.State != models.RunningStatus {
+			logger.Info("User creation job is scheduled")
+			r.EventRecorder.Eventf(o, models.Normal, models.CreationFailed,
+				"User creation job is scheduled, cluster is not in the running state",
+			)
+			return nil
+		}
+
+		for _, ref := range o.Spec.UserRefs {
+			err = r.createUser(ctx, logger, o, ref)
+			if err != nil {
+				logger.Info("Failed to create user on the cluster",
```

Comment:
`Failed to create a user on the cluster` -> `Failed to create a user for the cluster`

Reply:
fixed